### ISSUE TITLE
Blue Portals checklist

### ISSUE THREAD

**Issue by Luke18033** [BEEmod/BEE2-items#2139]
**The BP developers originally requested that we wait until Blue Portals: Anniversary Edition is released to use the custom assets. However, Anniversary Edition is now indefinitely on hold and we're not sure what this means for us. As such, Blue Portals content in BEE is not currently being worked on and we're not sure if it will ever release.**

## New items

- [ ] Fire Cube
- [ ] Ice Cube
- [ ] Unfizzlable Cube
	- Absolute Fizzlers and goo would still destroy it.
- [ ] Activation Cube
- [ ] Light Cube
	- Appears in Extended Play
	- Charges when hit by light, and emits until the charge runs out.
	- In P2 styles this would appear as a reflection cube with a purple skin or render color
- [ ] Magnet
- [ ] Repulsor
- [ ] Cube Activator
- [ ] Light
- [ ] Eye (light sensor)
- [ ] Mirror
- [ ] Cube Catcher
- [ ] Red Flow 
	- One voxel of liquid used for transport
	- Would use fizzler logic to expand to opposite surface, with no handle
- [ ] Water
	- Can be electrified using an Electric Panel
	- Always deadly in Co-op to avoid death animation issue
- [ ] Electric Panel
	- Replace with Still Alive zapper?
- [ ] Frozen Buttons
	- Would be placed on top of an existing button, adding the ice layer. This allows the item to be used with any button variant.
- [ ] ~Permanent Buttons~
	- Portal Gun On/Off buttons are already blue, so this would be confusing. Replace with a visible SR latch, which can be used with a normal button for this purpose.
- [ ] Visible Timer
- [ ] Fan (suggested in #1137)
- [ ] Hyper Cube Indicator

The following items may be featured in the Anniversary Edition, but are otherwise featured on a bit of wall graffiti or development material and will need to be custom made.
- [ ] Music Cube
	- Texture appears in the files, used in the cancelled sequel
- [ ] Timer Cube
	- Texture appears in the files, seen in an unreachable area, used in the cut time trial mode
- [ ] Swag Cube 
	- Appears on a wall, texture appears in the files
- [ ] Heavy Cube 
	- Appears on a wall, no in-game texture

### BPDC Style

Because this will be based on Anniversary Edition which has not released yet, the exact appearance of the style isn't known, though it will be ported.

- [ ] Style

### Misc

- [ ] Stylevar for Blue Portals portal gun and sounds
- [x] Blue Portals music (implemented)
- [ ] Modify standard Fizzler and Death Fizzler to exclude Unfizzlable Cubes
* Heavy Cube has a potential Old Aperture model from PS:Mel.

## Additional notes
1. ~P2 Style fire cubes should use an orange ring instead of a blue ring.~ No, the ring still needs to change color based on activated state, and would be colorisable.
2. Old Aperture variants of special cubes should have specially colored corners. The faces should be printed/painted with whichever emblem is on their BP counterparts.
3. ~Blue Portals signages will be appended to the BEESigns item or made into a new Blue Portal Signs item, depending on which and how many new items are implemented.~ No longer necessary, these will just be included in the customize signage menu.
4. BPDC style versions of default signages would have color added for fizzlers, fire, energy, etc. Inversely, BP signages would have color removed for other styles.
5. Soundtrack should be packaged with the mod, like Rexaura.
6. Music cubes should have matching music cube receptacles (determined by Button Type). 
7. Music cubes should have different tracks for each style, each two or four measures long.
	- Ideally this would be based on the selected music track rather than the style, though that would be a lot of work to make all of those. The app should support this functionality at the very least, falling back to the default style version if not present.
8. ~Wall cutout tiles should create the stone surfaces as seen in some chambers.~ Depends on Anniversary Edition style.
9. Fire, Ice, Light, Unfizzlable, and Heavy cubes should be bundled together with Cube Type. The Activation Cube would remain separate, as it supports connections while the others don't.
10. Unfizzlable Cubes should still be fizzled by Absolute Fizzlers and Goo.
	- Absolute Fizzlers are a more powerful fizzler, so it makes sense that they should affect these and it would be useful to prevent bringing them between chambers.
	- While the cubes could float in Goo, this doesn't have much use and would break with bottomless pits.

### Style Notes
* The Entrance Room would be a standard room that has a blue portal open, which leads to the entrance corridor.
* The exit room would be a corridor that leads to a door, that leads to the entrance room, but the blue portal is closed.
* The materials will be extracted from Anniversary Edition's files, when it is released.
* The style will be based off P1 style, meaning unstyled items will use their P1 versions.

**Comment by Byzarru** [comment]
I also had some ideas for technical names.

Standard Name | Technical Name
----------------- | ------------------
Fire Cube | Thermal Expulsion Cube
Ice Cube | Thermal Induction Cube
Unfizzlable Cube | Emancipation Resistance Cube
Light Cube | Spectrum Emission Cube
Heavy Cube | Theft-Proof Freight Cube

**Comment by Luke18033** [comment]
Some additional notes from @Byzarru:
1. P2 Style fire cubes should use an orange ring instead of a blue ring.
2. Old Aperture variants of special cubes should have specially colored corners. The faces should be printed/painted with whichever emblem is on their BP counterparts.
3. Blue Portals signages will be appended to the BEESigns item or made into a new Blue Portal Signs item, depending on which and how many new items are implemented. 
4. I will begin retexturing of the signage items (Connections and BEE especially) for BPDC style. Some icons are colored (Will color fizzlers, fire, energy). 
5. Soundtrack should be packaged, like with Rexaura.
6. Music cubes should have matching music cube receptacles (determined by Button type). 
7. Music cubes should have different tracks for each style, each two or four measures long (I might be able to make those sounds, I'll post them if I do)
8. Wall cutout tiles should create the stone surfaces as seen in some chambers.
9. Fire, Ice, Light, Unfizzlable, and Heavy cubes should be bundled together with Cube Type

**Comment by Luke18033** [comment]
BPDC Style notes from @TheDarkBomber:
* The Entrance Room would be a standard room that has a blue portal open, which leads to the entrance corridor.
* The exit room would be a corridor that leads to a door, that leads to the entrance room, but the blue portal is closed.
* The materials can be found in the game files for Blue Portals. Unportalable walls will use the stone texture, and Decorations would add white tiles on the walls. Portalable walls will be an orange version of P1 walls, as seen in the game.

**Comment by Luke18033** [comment]
There was also a BP variant checklist, but I don't think that's really necessary to copy over.

**Comment by Luke18033** [comment]
Red Flow should use the fizzler class, so that it extends to the opposite surface.

**Comment by Luke18033** [comment]
Countdown Timer is now implemented

**Comment by Luke18033** [comment]
Technically we can implement stuff like the Cube Catcher or Water without the permission, since they don't make use of any custom BP assets.

**Comment by Luke18033** [comment]
So, when exactly is the anniversary edition releasing?

**Comment by SW50000** [comment]
I don't think think they have a Release Date yet.

**Comment by Endermage77** [comment]
Personally, I would like to have a stylevar for the smiley-face antline/indicators, because I feel that smiley faces everywhere would not make very much sense, nor would it look very tidy in chambers.

**Comment by Luke18033** [comment]
Have you even played the actual mod? The smiley face icons appear in place of checkbox indicators, the actual antlines are just solid lines. I think BP also uses 16-unit wide antlines, and even if it doesn't they'd look fine at that size so we won't need to do any weird tricks with making the texture wider.

**Comment by Luke18033** [comment]
From the trailers it looks like Anniversary Edition uses regular P1 antlines, so if you really don't like the smiley-face icons you can use that instead.

**Comment by Endermage77** [comment]
Oh. Alright, sorry.

**Comment by Byzarru** [comment]
From what I understood, we are't going to be allowed to use the original textures and models anyway, the rHetorical team wants us to use the Anniversary Edition style.

**Comment by Luke18033** [comment]
I assumed we'd be able to use both once the anniversary edition is out. I don't like the new style, I prefer the original design.

**Comment by IceTudor** [comment]
Would water be placed like goo/bottomless/fire pits? (for firepits see https://github.com/BEEmod/BEE2-items/issues/2295)

**Comment by SW50000** [comment]
@IceTudor I don't think it would, water doesn't kill the player when you jump into it.
Goo would be the Coolant material seen in the mod.
![image](https://user-images.githubusercontent.com/20195894/31918120-9dac24e8-b820-11e7-9a64-5235e07db699.png)

**Comment by Byzarru** [comment]
Water would be placed like goo pits, it wouldn't replace the goo pits. Yes, the coolant texture would replace goo proper.

**Comment by TeamSpen210** [comment]
We can't make a custom goo item. I'd probably want a emitter thing you submerge, which then converts the attached goo into water.

**Comment by Endermage77** [comment]
You could do water channels that are placed like conveyor belts that you can string together, maybe?

**Comment by King0191** [comment]
Would the Light work like a BP-Styled Laser and the Mirrors would be a rotating plaform with a  reskinned Reflection Gel type thing? Or does it work differently?

**Comment by somedoodest** [comment]
I think this should have two styles, being the Original BPDC and Blue Portals 2 styles. In the BP2 style, Propulsion Gel would be replaced with oil, or maybe, for both BPDC styles, Oil will be an alternate version of Propulsion Gel.

**Comment by somedoodest** [comment]
Oh. Well, that idea is gone. But the BP2 style idea may be something you could look into.

**Comment by somedoodest** [comment]
Here are a few more ideas:
- Reflection cubes should be able to redirect Light (Could be controlled by Item Property.)
- Light cubes should be able to charge up with a Thermal Discouragement Beams (Could be controlled by Item Property).
- Item Property that allows Ice Cubes to freeze Goo / Coolant. 
- Timer Cube would act like in the original Time Trials, adding time to any active Neurotoxin Timers.
- Hyper Cube 3.0 could be added, controlled by Button Type. (Weighted = Regular, Cube = 3.0 Standalone Cube, Sphere = 3.0 Standalone Stand)
- the BP Cubes not seen in the mod would all be tied together, controlled by Cube type. (Normal = Timer Cube, Companion = Swag Cube, Reflection-Sphere-Frankenturret = Music Cubes #1,#2 and #3.)

**Comment by somedoodest** [comment]
Here are some missing items that I remember:
- Frozen Buttons
- Permanent Button
- Sphere Launcher (Could be changed in order to launch spheres automatically)
Also, Cube Catchers should be able to accept inputs, in the mod, there was a Cube Catcher that was opened by a frozen button.

**Comment by Luke18033** [comment]
I don't remember a "sphere launcher", could you upload a screenshot? Frozen and permanent buttons are missing from the list, I'll add those.

**Comment by Luke18033** [comment]
Although I'm not sure what the permanent buttons should look like, since one of the portalgun toggle buttons is already blue. Maybe it would be better to just have a visible version of the SR latch, which could be used to indicate that?

**Comment by somedoodest** [comment]
Yeah, your SR Latch to make Permanent Buttons idea is pretty good. Now here are screenshots of that sphere launcher I mentioned:
![obraz](https://user-images.githubusercontent.com/34423155/56848628-9355fc80-68eb-11e9-930a-0158fbaf5b6d.png)
![obraz](https://user-images.githubusercontent.com/34423155/56848652-bf717d80-68eb-11e9-9a5d-3dffac94c17e.png) (I had to stand close to that thing to do this screenshot, that's why it's so weird.)
Basically, the sphere launcher could be changed so it would have a Conveyor Belt leading up to the portalable surface. Now, I want this to be added as a new item, because of the red blocking field. Maybe that could be made into its own item, and the sphere launcher wouldn't be an item, only the field?

**Comment by somedoodest** [comment]
Also, I regret calling it a "launcher", it's more a surface to aim at.

**Comment by Luke18033** [comment]
Couldn't you replicate that by just using one of the custom fizzlers, like a CSF?

**Comment by somedoodest** [comment]
Yeah. It's really stupid. Also, I've created some textures for the BP cubes that, with a bit of modification, could be used. I could send those over.

**Comment by PlumbingFan** [comment]
I have an Idea for the Water Electric Panel/Electric Water: there could be an option to set it to "Forgiving" Like on the Lazerfields were it will kill you in X amount of hits/time

**Comment by Anus95** [comment]
Hows it going so far? whats done and whats still to do

**Comment by Flash4433** [comment]
Check the first message of this issue. They have permission from the mod creator to use the their items, but the mod creator requested that they wait until Anniversary Edition is released to use the assets.

> We currently have permission from Rhetorical to use their items, but they have requested that we wait until Anniversary Edition is released to use the assets. This means the BP style won't be able to start development until a later point in time, and we'll need to make our own assets for the items in other styles (which we'd do anyway).

**Comment by somedoodest** [comment]
Well, this is basically dead unless the permission to use the items still applies to BP: Redux. Which i kinda doubt.

**Comment by somedoodest** [comment]
I'll just leave a few ideas for improved / new technical names for cubes and other items inspired by Byzarru's original post, even if this is dead.

Standard Name | Technical Name
-- | --
Light Cube | Luminescence Storage Cube
Timer Cube | Timer Prolongation Cube
Music Cube | Melody Emission Cube
Magnet | Magnetic Attraction Device
Repulsor | Magnetic Repulsion Device
Cube Activator | Emancipating Trigger Device
Light | Spectrum Emission Device
Eye (Light sensor) | Spectrum Detection Device
Cube Catcher | Cube Identification Device
Red Flow | Liquid Suspension Device

**Comment by Luke18033** [comment]
Honestly, do we really need technical names for the test elements? Most people probably won't use them anyway.

**Comment by Endermage77** [comment]
Speaking of, I haven't heard much about the status of the mod whose release these items hinge upon for being implemented.

**Comment by somedoodest** [comment]
Well, it's probably dead, considering BP: Redux's existence. If not, then we'll probably get BP:AE for the 10th anniversary later this year.

**Comment by Luke18033** [comment]
I thought BP:Redux was just a different name for Anniversary Edition? Either way, we'd just base on that version instead.

**Comment by somedoodest** [comment]
Nope, Redux is a new BP project that uses Portal 2 (due to source code access). It'll be cool seeing the new "stylized" versions of everything once that drops.

**Comment by Luke18033** [comment]
Hmm, that might cause problems for us if they decide to hardcode all the custom test elements. We could probably just use the original logic though, or new custom VScripts. I'm not really counting on Redux releasing though, given how many other cancelled BP-related projects there have been. I can barely even find any info about it online (this very issue is one of the top Google search results for "blue portals redux").

**Comment by somedoodest** [comment]
Look on Blue Portals' mod page on ModDB. That's where they revealed it and responded to a few comments. There was also an additional image posted on twitter by Colossal / Chris Weyland.

**Comment by somedoodest** [comment]
Here's an idea for a "Maintenance" style (if there will be any in Redux / AE's potential release later this year):

- The Entrance Room should look like the Exit Room of Chamber 9 / Entrance Room of Chamber 10
- The Exit Room should be the same as the Entrance Room
- Observation Rooms on the ceiling should generate a hole with a view into a Vactube, which would sometimes transport BP Cubes / the regular cubes inside of it
- Use #414 to create things like police tape and random junk on the ground

**Comment by somedoodest** [comment]
And for good measure, some extra ideas.
- BP styles could have both Regular and Maintenance Entrance Rooms and Exit Rooms with settings in the Corridor section of the "Compile Options" window.
- Visible Timers should be an option for Neurotoxin Timers.
- If Redux will be the base for most of the items/styles, then Portal 1 variants of items should use the textures from the original Blue Portals.
- The "Cube Gels" from "Conversion Cubed" could be items.
- Cube Catchers should be able to identify "Crap", perhaps Cube Type could dictate what it considers "Crap"?
- Perhaps one of the Maintenance style's exit corridors could be the hallway seen in the ending of the original Blue Portals.

**Comment by somedoodest** [comment]
Here's a request: Please start developing the items whenever Anniversary Edition's files release. It would be a shame to wait until 202X (yes, that is Redux's release date currently.) to start developing items that could have been done prematurely using what was supposed to be the OG base for the items.

**Comment by Luke18033** [comment]
I guess if they said to "wait until Anniversary Edition is out", then Reepblue posting the files online technically counts...

**Comment by somedoodest** [comment]
Question: can you change the Timer Cube's note to "Texture appears in the files, replaced regular cube in the removed Time Trials mode"? It was used in Time Trials as the replacement for the regular cube before that mode got completely removed from the mod. 
There were also some timer variants of weighted buttons and pedestal buttons, which would become separate items that add time to timers themselves.

**Comment by Luke18033** [comment]
I think the plan was to use that as a timed variant of the Activation Cube, since the original time trial mode was cut. What did the timer cube do anyway, give extra time when picked up?

**Comment by somedoodest** [comment]
It didn't actually give time, it was a reskin of the regular cube in Time Trials and the only thing that added time in Time Trials.. were special pedestal buttons. I personally think this version should add time when put on Weighted / Cube buttons.

**Comment by somedoodest** [comment]
uh. BP:AE is not canceled, apparently. it's between devs. (offical responce from Colossal, btw)

**Comment by maxdevgithub** [comment]
There must be a box which wont work for Box Catcher aka c**p box as seen in reepblues hammer notepad

**Comment by maxdevgithub** [comment]
BP Pre Anniversary would be ok

**Comment by Luke18033** [comment]
Except that the BP developers said that they don't want people using the original assets. But I'll probably ask about it again at some point, it doesn't seem like Anniversary Edition is really getting anywhere.

**Comment by maxdevgithub** [comment]
> Check the first message of this issue. They have permission from the mod creator to use the their items, but the mod creator requested that they wait until Anniversary Edition is released to use the assets.
> 
> > We currently have permission from Rhetorical to use their items, but they have requested that we wait until Anniversary Edition is released to use the assets. This means the BP style won't be able to start development until a later point in time, and we'll need to make our own assets for the items in other styles (which we'd do anyway).

As luke said, it doesn't seem like Anniversary Edition is really getting anywhere. He will ask again sometime

**Comment by Luke18033** [comment]
What? You were the one who asked about it originally.

**Comment by maxdevgithub** [comment]
Well i did ask Pre-AE style, if you can of course. I was just replying. Due to BPAE not releasing in 2018 some may be impatient.

**Comment by somedoodest** [comment]
hey luke, think you could get in contact with the devs soon? Colossal said Anniversary Edition is on hold indefinetly during Reep's recent BP livestream.

**Comment by Luke18033** [comment]
Ah. TeamSpen originally contacted them by email, he'll probably want to ask about it again.

**Comment by supersean08-pro** [comment]
ay guys not to worry i made that cube detector BP thing

**Comment by Bloodstrike2003** [comment]
Interesting

**Comment by Luke18033** [comment]
I'm not really sure what the status of this is, closing it for now.

**Comment by supersean08-pro** [comment]
> I'm not really sure what the status of this is, closing it for now.

It aint coming man
![Here](https://user-images.githubusercontent.com/63530071/235333008-8c1acd44-482c-4c98-8fd7-8f265c6ae1cc.png)
